File: examples/javascript/connect_sandbox.js
/**
 * Connect to existing sandbox - Reconnect and interact.
 *
 * Usage:
 *   E2B_DOMAIN=localhost:8080 E2B_API_KEY=test-key node connect_sandbox.js
 */

const { Sandbox } = require("@e2b/code-interpreter");

async function main() {
  const apiKey = process.env.E2B_API_KEY || "test-key";
  const domain = process.env.E2B_DOMAIN || "localhost:8080";

  // 1. Create a sandbox
  console.log("1. Creating sandbox...");
  const sandbox = await Sandbox.create({ apiKey, domain });
  const sandboxId = sandbox.sandboxId;
  console.log(`   Created: ${sandboxId}`);

  // 2. Write some state
  console.log("2. Writing state...");
  await sandbox.commands.run("echo 'persistent state' > /tmp/state.txt");
  await sandbox.kill();
  console.log(`   Disconnected from: ${sandboxId}`);

  // 3. List running sandboxes
  console.log("3. Listing sandboxes...");
  const sandboxes = await Sandbox.list({ apiKey, domain });
  console.log(`   Running: ${sandboxes.length}`);

  // 4. Connect to existing sandbox
  console.log("4. Reconnecting to sandbox...");
  try {
    const reconnected = await Sandbox.connect(sandboxId, { apiKey, domain });
    const result = await reconnected.commands.run("cat /tmp/state.txt");
    console.log(`   State: ${result.stdout.trim()}`);
    await reconnected.kill();
  } catch (e) {
    console.log(`   Connect failed (expected for mock): ${e.message}`);
  }

  console.log("\nDone.");
}

main().catch(console.error);

File: examples/python/connect_sandbox.py
"""Connect to existing sandbox - Reconnect and interact.

Demonstrates connecting to a running sandbox:
1. Create a sandbox
2. Disconnect (keep running)
3. Reconnect by sandbox ID
4. Interact with the reconnected sandbox

Usage:
    export E2B_DOMAIN=localhost:8080
    export E2B_API_KEY=your-api-key
    python connect_sandbox.py
"""

import os
from e2b import Sandbox


def main():
    api_key = os.environ.get("E2B_API_KEY", "test-key")
    domain = os.environ.get("E2B_DOMAIN", "localhost:8080")

    # 1. Create a sandbox
    print("1. Creating sandbox...")
    sandbox = Sandbox.create(
        template="base",
        api_key=api_key,
        domain=domain,
    )
    sandbox_id = sandbox.sandbox_id
    print(f"   Created: {sandbox_id}")

    # 2. Write some state
    print("2. Writing state...")
    sandbox.commands.run("echo 'persistent state' > /tmp/state.txt")
    sandbox.kill()
    print(f"   Disconnected from: {sandbox_id}")

    # 3. List running sandboxes to find it
    print("3. Listing sandboxes...")
    sandboxes = Sandbox.list(api_key=api_key, domain=domain)
    print(f"   Running: {len(sandboxes)}")

    # 4. Connect to existing sandbox
    # Note: This requires the sandbox to still be running
    # In real E2B, you'd use Sandbox.connect(sandbox_id)
    print("4. Reconnecting to sandbox...")
    try:
        reconnected = Sandbox.connect(
            sandbox_id,
            api_key=api_key,
            domain=domain,
        )
        result = reconnected.commands.run("cat /tmp/state.txt")
        print(f"   State: {result.stdout.strip()}")
        reconnected.kill()
    except Exception as e:
        print(f"   Connect failed (expected for mock): {e}")
        print("   Creating new sandbox to verify...")

    print("\nDone.")


if __name__ == "__main__":
    main()
